graphene_sqlalchemy/batching.py

```diff
@@ -62,26 +70,42 @@ async def batch_load_fn(self, parents):
                 query_context = parent_mapper_query._compile_context()
 
             if is_sqlalchemy_version_less_than('1.4'):
-                selectin_loader._load_for_path(
+                self.selectin_loader._load_for_path(
                     query_context,
                     parent_mapper._path_registry,
                     states,
                     None,
                     child_mapper
                 )
             else:
-                selectin_loader._load_for_path(
+                self.selectin_loader._load_for_path(
                     query_context,
                     parent_mapper._path_registry,
                     states,
                     None,
                     child_mapper,
                     None
                 )
-
-            return [getattr(parent, relationship_prop.key) for parent in parents]
-
-    loader = RelationshipLoader()
+            return [getattr(parent, self.relationship_prop.key) for parent in parents]
+
+    def _get_loader(relationship_prop):
+        """Retrieve the cached loader of the given relationship."""
+        loader = RELATIONSHIP_LOADERS_CACHE.get(relationship_prop, None)
+        if loader is None:
+            selectin_loader = strategies.SelectInLoader(
+                relationship_prop,
+                (('lazy', 'selectin'),)
+            )
+            loader = RelationshipLoader(
+                relationship_prop=relationship_prop,
+                selectin_loader=selectin_loader
+            )
+            RELATIONSHIP_LOADERS_CACHE[relationship_prop] = loader
+        else:
+            loader.loop = get_event_loop()
```

Comment:
I'm not 100% sure about this. Do any of you see issues with this?

Reply:
Why do we need the get_event_loop() call in the else part? Isn't that initially covered by `aiodataloader` already, and shouldn't the cache also get a full reset if the event loop changes (implying thread restart/change)?

`aiodataloader` recommends to create a new loader for each request, this would keep the loaders between different queries, right?

Reply:
Makse sense, I changed it

Reply:
September update: There is also preliminary work in Graphene to vendor the `DataLoader` from `aiodataloader` into Graphene, and also change it's `__init__` behavior to not ask for an even loop so early on init. Just warning because that might touch the same points we are touching here.